### PR TITLE
Feat/multi task classification

### PR DESCRIPTION
**Pull request by Optimox**
**What kind of change does this PR introduce?**

This adds a new class for multi task multi classification problem TabNetMultiTaskClassifier.

Still WIP as documentation need to be added.

**Does this PR introduce a breaking change?**
There should be no breaking changes, just new features.

**What needs to be documented once your changes are merged?**
new class, a few differences in behaviour, like being able to give a list of loss function and not only one loss function

**Closing issues**

Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if such).

### CODE REVIEW

File: pytorch_tabnet/multitask.py [hunk 1]
```diff
@@ -0,0 +1,348 @@
...
+        if isinstance(self.loss_fn, list):
+            # if you specify a different loss for each task
+            for task_loss, task_output, task_id in zip(self.loss_fn,
+                                                       output,
+                                                       range(len(self.loss_fn))):
+                loss += task_loss(task_output, targets[:, task_id])
+        else:
+            # same loss function is applied to all tasks
+            for task_id, task_output in enumerate(output):
+                loss += self.loss_fn(task_output, targets[:, task_id])
```
Comment (eduardocarvp) [inline pytorch_tabnet/multitask.py:186]: Maybe something like:
```
if isinstance(self.loss_fn, list):
    losses = self.loss_fn
else:
    # same loss function is applied to all tasks
    losses = [self.loss_fn] * len(output)

for task_id, task_loss, task_loss in enumerate(zip(losses, output)):
    loss += self.loss_fn(task_output, targets[:, task_id])
``` 

Not sure if it's any easier to understand though...

File: pytorch_tabnet/utils.py [hunk 2]
```diff
@@ -168,3 +168,25 @@ def create_explain_matrix(input_dim, cat_emb_dim, cat_idxs, post_embed_dim):
...
+
+
+def filter_weights(weights):
+    """
+        This function makes sure that weights are in correct format for
+        regression and multitask TabNet
+
+    Parameters
+    ----------
+    weights: int, dict or list
```
Comment (eduardocarvp) [inline pytorch_tabnet/utils.py:180]: Weights cannot be a dictionary here

**Review by eduardocarvp — APPROVED** [review]
